Library/Homebrew/language/haskell.rb

```diff
@@ -63,8 +67,20 @@ def install_cabal_package(*args)
         cabal_sandbox do
           cabal_install_tools(*options[:using]) if options[:using]
 
+          # if we have build flags, we have to pass them to cabal install to resolve the necessary
+          # dependencies, and call cabal configure afterwards to set the flags again for compile
+          flags = ""
+          if options[:flags]
+            flags = "--flags='#{options[:flags].join(" ")}'"
+          end
+
+          args_and_flags = [*args, flags]
```

Comment:
`*args` needs to be last on Ruby 1.8. You could build the array in two steps instead.